source/common/http/match_delegate/config.cc

```diff
@@ -135,6 +147,16 @@ DelegatingStreamFilter::DelegatingStreamFilter(
 
 Envoy::Http::FilterHeadersStatus
 DelegatingStreamFilter::decodeHeaders(Envoy::Http::RequestHeaderMap& headers, bool end_stream) {
+  const auto* per_route_config =
+      Envoy::Http::Utility::resolveMostSpecificPerFilterConfig<FilterConfigPerRoute>(
+          decoder_callbacks_);
+
+  if (per_route_config != nullptr) {
+    match_state_.setMatchTree(per_route_config->matchTree());
+  } else {
+    ENVOY_LOG(debug, "No per route config found");
```

Comment:
nit: let's reduce it from `debug` to `trace` since it is ok we don't have per route config. i.e. matcher can come from filter config OR no matcher at all (skip).

and let's add a bit more details :  "No per route config found, thus the matcher tree (if exists) will not be built from per route config".

Reply:
Yes, this actually will be triggered  when there is no per route config (per route matcher is part of per route config),
which is normal case. So that is why I want to reduce it to `trace`